### ISSUE TITLE
Use before definition errors should not be reported in ambient contexts

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#9782]
```ts
// test.d.ts

declare var S: typeof A; // error TS2448: Block-scoped variable 'A' used before its declaration.
declare const A: number;  
```

Errors make sense only in non-ambient context.

**Comment by mihailik** [comment]
But would that error still be valid like this?

```typescript
var S: typeof A;
const A: number;
```

I would argue we're not **using** A variable, we're reasoning about its type. And the type is known regardless of the order of the statements.
